Parser.cpp

```diff
@@ -143,7 +143,7 @@ std::optional<std::string> commonItems::parser::getNextToken(std::istream& theSt
 			match->second(toReturn, theStream);
 			matched = true;
 		}
-		else if (toReturn.size() > 2 && toReturn[0] == '\"' && toReturn[toReturn.size() - 1] == '\"')
+		else
```

Comment:
No, no. Same as below. Firing up a regex match costs far more than checking it if it's worth it. And if there's no quotes it's not worth it, and you'll be firing it for every single match in existance. You may want to make a small bool function in stringutils isStringQuoted, so you only fire up regex here and below if it is.

Reply:
We really should tiptoe around parser as if walking on eggs.

Reply:
See dcd3ec7, for now I moved `strippedLexeme = remQuotes(toReturn);` up to remove duplication and brought back the ifs. We've gone beyond the scope of the PR already.

Reply:
that is good, now you can make it even cheaper by replacing:
(toReturn.size() > 2 && toReturn[0] == '\"' && toReturn[toReturn.size() - 1] == '\"')
with:
stippledLexeme.size() < toReturn.size()

Reply:
This is a really good idea, I added an `isLexemeQuoted` bool to make that comparison only once.